src/hb-paint.h

```diff
@@ -0,0 +1,785 @@
+ * @start_angle: the start angle, in radians
+ * @end_angle: the end angle, in radians
+ * @user_data: User data pointer passed to hb_paint_funcs_set_sweep_gradient_func()
+ *
+ * A virtual method for the #hb_paint_funcs_t to paint a sweep
+ * gradient everywhere within the current clip.
+ *
+ * The coordinates of the points are interpreted according
+ * to the current transform.
+ *
+ * See the OpenType spec [COLR](https://learn.microsoft.com/en-us/typography/opentype/spec/colr)
+ * section for details on how the points define the direction
+ * of the gradient, and how to interpret the @color_line.
+ *
+ * Since: REPLACEME
+ */
+typedef void (*hb_paint_sweep_gradient_func_t)  (hb_paint_funcs_t *funcs,
+                                                 void *paint_data,
+                                                 hb_color_line_t *color_line,
+                                                 float x0, float y0,
+                                                 float start_angle,
+                                                 float end_angle,
+                                                 void *user_data);
+
+/**
+ * hb_paint_composite_mode_t:
+ *
+ * The values of this enumeration describe the compositing modes
+ * that can be used when combining temporary redirected drawing
+ * with the backdrop.
+ *
+ * See the OpenType spec [COLR](https://learn.microsoft.com/en-us/typography/opentype/spec/colr)
+ * section for details.
+ */
+typedef enum {
+  HB_PAINT_COMPOSITE_MODE_CLEAR,
+  HB_PAINT_COMPOSITE_MODE_SRC,
+  HB_PAINT_COMPOSITE_MODE_DEST,
+  HB_PAINT_COMPOSITE_MODE_SRC_OVER,
+  HB_PAINT_COMPOSITE_MODE_DEST_OVER,
+  HB_PAINT_COMPOSITE_MODE_SRC_IN,
+  HB_PAINT_COMPOSITE_MODE_DEST_IN,
+  HB_PAINT_COMPOSITE_MODE_SRC_OUT,
+  HB_PAINT_COMPOSITE_MODE_DEST_OUT,
+  HB_PAINT_COMPOSITE_MODE_SRC_ATOP,
+  HB_PAINT_COMPOSITE_MODE_DEST_ATOP,
+  HB_PAINT_COMPOSITE_MODE_XOR,
+  HB_PAINT_COMPOSITE_MODE_PLUS,
+  HB_PAINT_COMPOSITE_MODE_SCREEN,
+  HB_PAINT_COMPOSITE_MODE_OVERLAY,
+  HB_PAINT_COMPOSITE_MODE_DARKEN,
+  HB_PAINT_COMPOSITE_MODE_LIGHTEN,
+  HB_PAINT_COMPOSITE_MODE_COLOR_DODGE,
+  HB_PAINT_COMPOSITE_MODE_COLOR_BURN,
+  HB_PAINT_COMPOSITE_MODE_HARD_LIGHT,
+  HB_PAINT_COMPOSITE_MODE_SOFT_LIGHT,
+  HB_PAINT_COMPOSITE_MODE_DIFFERENCE,
+  HB_PAINT_COMPOSITE_MODE_EXCLUSION,
+  HB_PAINT_COMPOSITE_MODE_MULTIPLY,
+  HB_PAINT_COMPOSITE_MODE_HSL_HUE,
+  HB_PAINT_COMPOSITE_MODE_HSL_SATURATION,
+  HB_PAINT_COMPOSITE_MODE_HSL_COLOR,
+  HB_PAINT_COMPOSITE_MODE_HSL_LUMINOSITY,
+} hb_paint_composite_mode_t;
+
+/**
+ * hb_paint_push_group_func_t:
+ * @funcs: paint functions object
+ * @paint_data: The data accompanying the paint functions in hb_font_paint_glyph()
+ * @user_data: User data pointer passed to hb_paint_funcs_set_push_group_func()
+ *
+ * A virtual method for the #hb_paint_funcs_t to use
+ * an intermediate surface for subsequent paint calls.
+ *
+ * The drawing will be redirected to an intermediate surface
```

Comment:
Who will be doing this redirection, HarfBuzz (as implied by the wording) or the callback function (which I think what is meant here)?